state-chain/chains/src/btc/utxo_selection.rs

```diff
@@ -19,26 +19,27 @@ use super::{ConsolidationParameters, GetUtxoAmount};
 ///
 /// Note that on failure (when `None` is returned) this may still modify `available_utxos`, which
 /// is not a concern because the caller will ignore that value in that case.
-pub fn select_utxos_from_pool<UTXO: GetUtxoAmount + Clone>(
-	available_utxos: &mut Vec<UTXO>,
-	fee_per_utxo: u64,
+pub fn select_utxos_from_pool(
+	available_utxos: &mut Vec<Utxo>,
+	fee_info: &BitcoinFeeInfo,
 	amount_to_be_spent: u64,
-) -> Option<(Vec<UTXO>, u64)> {
+) -> Option<(Vec<Utxo>, u64)> {
 	if available_utxos.is_empty() {
 		return None
 	}
 
-	available_utxos.sort_by_key(|utxo| sp_std::cmp::Reverse(utxo.amount()));
+	available_utxos.sort_by_key(|utxo| sp_std::cmp::Reverse(utxo.amount));
 
-	let mut selected_utxos: Vec<UTXO> = vec![];
-	let mut skipped_utxos: Vec<UTXO> = vec![];
+	let mut selected_utxos: Vec<Utxo> = vec![];
+	let mut skipped_utxos: Vec<Utxo> = vec![];
 
 	let mut cumulative_amount = 0;
 
 	while cumulative_amount < amount_to_be_spent {
 		if let Some(current_smallest_utxo) = available_utxos.pop() {
-			if current_smallest_utxo.amount() > fee_per_utxo {
-				cumulative_amount += current_smallest_utxo.amount() - fee_per_utxo;
+			if current_smallest_utxo.amount > fee_info.fee_for_utxo(&current_smallest_utxo) {
```

Comment:
`fee_info.fee_for_utxo(&current_smallest_utxo)` could be a variable and deduplicated

Reply:
done